node/sc/mainbridge_test.go

```diff
@@ -195,3 +239,113 @@ func TestMainBridge_handleMsg(t *testing.T) {
 	}
 	_ = pipe1.Close()
 }
+
+// TestMainBridge_handle tests the fail cases of `handle` function.
+// There are no success cases in this test since `handle` has a infinite loop inside.
+func TestMainBridge_handle(t *testing.T) {
+	// Create a MainBridge
+	mBridge := testNewMainBridge(t)
+	defer mBridge.chainDB.Close()
+
+	// Set testBlockChain to MainBridge.blockchain
+	bc, err := testBlockChain()
+	if err != nil {
+		t.Fatal(err)
+	}
+	mBridge.blockchain = bc
+
+	// Variables will be used as return values of mockBridgePeer
+	key, _ := crypto.GenerateKey()
+	nodeID := discover.PubkeyID(&key.PublicKey)
+	peer := p2p.NewPeer(nodeID, "name", []p2p.Cap{})
+	peerID := peer.ID()
+	bridgePeerID := fmt.Sprintf("%x", peerID[:8])
+	pipe, _ := p2p.MsgPipe()
+
+	// mockBridgePeer mocks BridgePeer
+	mockCtrl := gomock.NewController(t)
```

Comment:
You should call `mockCtrl.Finish()` at the end of the test or `defer mockCtrl.Finish()` to indicate the end of mocking test. If it is not called, test does not fail even if it does not meet the expectations.

Reply:
Thank you for the notice! I added it.